db/migrate/20141214220050_create_proposals.rb

```diff
@@ -0,0 +1,37 @@
+class CreateProposals < ActiveRecord::Migration
+  def change
+    create_table :proposals do |t|
+      t.string :status
+      t.string :flow
+      t.timestamps
+    end
+    add_reference :carts, :proposal
+
+    reversible do |dir|
+      Cart.reset_column_information
+
+      dir.up do
+        Cart.find_each do |cart|
+          cart.create_proposal!(
```

Comment:
Note that it's keeping the same `id` on the carts, so none of the other references need to change.